Ska/Shell/shell.py

```diff
@@ -152,8 +152,16 @@ def run_shell(cmdstr, shell='bash', logfile=None, importenv=False, getenv=False,
     shell.delaybeforesend = 0.0
 
     if env:
-        setenv_str = "export %s='%s'" if shell_name == 'bash' else "setenv %s '%s'"
+        exclude_vars = []
+        if shell_name == 'bash':
+            setenv_str = "export %s='%s'"
+            exclude_vars = ['PS1', 'PS2', 'PS3', 'PS4', 'PROMPT_COMMAND']
+        else:
+            setenv_str = "setenv %s '%s'"
+            exclude_vars = ['PROMPT', 'PROMPT2']
         for key, val in env.items():
+            if key in exclude_vars:
+                continue
```

Comment:
This should probably be more idiomatic.

Reply:
`if key not in exclude_vars:` would work.